src/Microsoft.ML.Data/Transforms/HashTransform.cs

```diff
@@ -28,19 +36,8 @@ namespace Microsoft.ML.Runtime.Data
     /// it hashes each slot separately.
     /// It can hash either text values or key values.
     /// </summary>
-    public sealed class HashTransform : OneToOneTransformBase, ITransformTemplate
+    public sealed class HashConverterTransformer : OneToOneTransformerBase
```

Comment:
>HashConverterTransformer [](start = 24, length = 24)

What is this hash *converter*?

I do not like this usage of "converter." You have "hash converter transformer". But to transform is also a form of conversion. So it's like you are saying "this is a hash transformer transformer" which makes no sense. What is wrong with "Hash Estimator" and "Hash Transformer"?

Reply:
>>1) namespace Microsoft.ML.Transforms
>>2) Estimators should be named the same as the Legacy ones as appropriate, with no Estimator suffix. In this case, it should be HashConverter.
>>3) Transformers should be named the same as estimators, with a Transformer suffix: HashConverterTransformer.
I'm bad with words, so I don't have strong opinion on this.

---
In reply to: [218939448](https://github.com/dotnet/machinelearning/pull/944#discussion_r218939448) [](ancestors = 218939448)

Reply:
Also I not is smithword. 
I thought that we should just adopt the naming of the legacy components, assuming that some thought has been put into naming these. Legacy has `HashConverter` as one step (it uses HashJoin, but I think it's irrelevant).

---
In reply to: [218942124](https://github.com/dotnet/machinelearning/pull/944#discussion_r218942124) [](ancestors = 218942124,218939448)

Reply:
The estimators and (sometimes) transformers are things directly instantiated by users. To be instantiated by users they must be found. In order to be reasonably found the best way I know of is that they have a uniform consistent naming policy, and a common policy that people understand in .NET that I've seen many, mnay frameworks adopt is that an implementor of interface `IFoo` or some base class `FooBase`. will have `Foo` somewhere in the name. This also jibes with how we've written our names everywhere else. E.g.: `IDataTransform` (nearly?) all transforms have `Transform` in their name.

---
In reply to: [218945320](https://github.com/dotnet/machinelearning/pull/944#discussion_r218945320) [](ancestors = 218945320,218942124,218939448)

Reply:
I agree with what you said. Still, I think there is value in preserving 'verb-like' nouns like 'ColumnConcatenator' and 'TextTokenizer'. Having something like `ColumnConcatenatorEstimator` sounds bad to me. Do you have a better suggestion for the naming policy?

---
In reply to: [218948931](https://github.com/dotnet/machinelearning/pull/944#discussion_r218948931) [](ancestors = 218948931,218945320,218942124,218939448)

Reply:
Sure. Any implementor of `IEstimator` should be suffixed with `Esimator`, and implementor of `ITransformer` ought to be suffixed with `Transformer`. This will make them easy to find.

"Sounds bad to me" is a little hard for me to address. Could I possibly convince you to be more specific? Is the problem that it's too long, or you don't like the two `-or` suffix words in a row? If too long, maybe shorten it. If it's the two `-or` suffix words in a row, then I'd suggest maybe making the first one not that. E.g., we can change `ColumnConcatenatorEstimator` to `ColumnConcatenatingEstimator` or `ColumnConcatenationEstimator` or something.

---
In reply to: [218975244](https://github.com/dotnet/machinelearning/pull/944#discussion_r218975244) [](ancestors = 218975244,218948931,218945320,218942124,218939448)

Reply:
I don't like that there are two `-or` suffixes, and that the name is long. 
`ColumnConcatenationEstimator` sounds better, but still long. 

I don't like the idea that every estimator should be suffixed with `Estimator`. What about trainers? Do we also suffix them with `Estimator`? But they are also trainers, so maybe `TrainerEstimator`?
`StochasticDualCoordinateAscentTrainerEstimator`, for example? we also need to throw in something about the task, so `StochasticDualCoordinateAscentMulticlassTrainerEstimator`.

---
In reply to: [219182371](https://github.com/dotnet/machinelearning/pull/944#discussion_r219182371) [](ancestors = 219182371,218975244,218948931,218945320,218942124,218939448)

Reply:
We should have a naming convention to distinguish among the different classes of objects.

---
In reply to: [219232649](https://github.com/dotnet/machinelearning/pull/944#discussion_r219232649) [](ancestors = 219232649,219182371,218975244,218948931,218945320,218942124,218939448)